src/libs/Pusher/pusher.ts

```diff
@@ -90,8 +93,9 @@ function callSocketEventCallbacks(eventName: SocketEventName, data?: EventCallba
  * @returns resolves when Pusher has connected
  */
 function init(args: Args, params?: unknown): Promise<void> {
-    initPromise = new Promise((resolve) => {
+    return new Promise((resolve) => {
```

Comment:
Good catch! It could lead to this sequence and crash:
1. `User A` logs in
2. `initPromise` is resolved
3. `User A` logs out
4. `socket` is reassigned to `null`
5. `User B` logs in
6. `initPromise` is resolved but `socket` is not created yet
7. App crashes here https://github.com/Expensify/App/blob/31adaa82ee9ee70cbcee296060cdebbe9f56fab5/src/libs/Pusher/pusher.ts#L249-L250

Reply:
But we can avoid this by creating new `Promise` in `disconnect` function like that:
```
/**
 * Disconnect from Pusher
 */
function disconnect() {
    if (!socket) {
        Log.info('[Pusher] Attempting to disconnect from Pusher before initialisation has occurred, ignoring.');
        return;
    }

    socket.disconnect();
    socket = null;
    pusherSocketID = '';
    initPromise = new Promise((resolve) => {
        resolveInitPromise = resolve;
    });
}
```

Reply:
Good catch! That would cause a bug as now if you call `pusher.subscribe()` it will fail with `[Pusher] instance not found`, socket is cleared on logout and not created again since the promise is already resolved.

Actually the bug already exists on main too as we can always call `subscribe` before `init`

Reply:
Github comments are not loading up in time 😅